src/components/modal.js

```diff
@@ -0,0 +1,207 @@
+/** @jsx jsx */
+import React, { Component } from 'react';
+import { jsx, Flex, Styled } from 'theme-ui';
+import ReactModal from 'react-modal';
+import { X, ArrowLeft } from 'react-feather';
+
+import IconButton from './icon-botton';
+import Text from './text';
+
+ReactModal.setAppElement('body');
+
+class ModalWrapper extends Component {
+  static defaultProps = {
+    isOpen: false,
+    contentLabel: '',
+    closeTimeoutMS: 300,
+    style: {},
+    contentStyle: {},
+    wrapperStyle: {},
+    width: 'fit-content',
+    height: 'fit-content',
+    scrollable: true,
+    centerTitle: false,
+    hideCloseButton: false,
+  };
+
+  state = {
+    contentSpacing: 0,
+  };
+
+  afterOpenModal = () => {
+    const headerHeight = this.headerRef ? this.headerRef.clientHeight : 0;
+    const footerHeight = this.footerRef ? this.footerRef.clientHeight : 0;
+    let contentSpacing = headerHeight + footerHeight;
+
+    this.setState({ contentSpacing });
+  };
+
+  renderBackButton = (onBack) => (
+    <IconButton
+      label="back"
+      onClick={onBack}
+      sx={{
+        position: 'absolute',
+        top: 45,
+        left: 20,
+        zIndex: 1,
+      }}
+    >
+      <ArrowLeft />
+    </IconButton>
+  );
+
+  renderCloseIcon = (onClose) => (
+    <IconButton
+      label="close"
+      onClick={onClose}
+      sx={{
+        position: 'absolute',
+        top: 45,
```

Comment:
Can we switch to using `space` values from the theme? https://github.com/ifings/metatron3/blob/master/src/components/theme/space.js

Reply:
It should read correctly based on the Theme UI spec: https://theme-ui.com/theme-spec

Reply:
I've updated it so that it uses the `space` values